mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -74,6 +74,55 @@ macro_rules! event {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct AuthScheme {
+    disable_double_encoding: bool,
+    scheme_name: String,
+    signing_name: String,
+    signing_region: String,
+}
+
+impl Default for AuthScheme {
```

Comment:
Is the builder pattern (and `Default`) needed here? It seems to me that a simple `new()` would work fine.